src/lib/chain/offer.ts

```diff
@@ -340,72 +444,176 @@ async function resolveOffersFromProviderConfig(flags: OffersArgs) {
+  ) {
+    commandObj.error(
+      `You can't use both ${color.yellow(
+        `--${OFFER_FLAG_NAME}`,
+      )} and ${color.yellow(
+        `--${OFFER_IDS_FLAG_NAME}`,
+      )} flags at the same time. Please pick one of them`,
+    );
+  }
+
+  if (flags[OFFER_IDS_FLAG_NAME] !== undefined) {
+    return commaSepStrToArr(flags[OFFER_IDS_FLAG_NAME]).map((offerId, i) => {
+      return { offerName: `#${i}`, offerId };
+    });
+  }
+
   const providerArtifactsConfig = await initReadonlyProviderArtifactsConfig();
 
   if (providerArtifactsConfig === null) {
-    return [];
+    commandObj.error(
+      `Wasn't able to find ${PROVIDER_ARTIFACTS_CONFIG_FULL_FILE_NAME} in ${DOT_FLUENCE_DIR_NAME} which means you probably didn't create any offers yet. Please run '${CLI_NAME} provider offer-create' command first`,
+    );
   }
 
-  const [notFoundOffers, offers] = splitErrorsAndResults(
-    flags.offers === undefined
-      ? Object.keys(providerArtifactsConfig.offers)
-      : commaSepStrToArr(flags.offers),
+  const allOffers: OfferFromProviderArtifacts[] = Object.entries(
+    providerArtifactsConfig.offers,
+  ).map(([offerName, { id }]) => {
+    return { offerName, offerId: id };
+  });
+
+  if (flags[OFFER_FLAG_NAME] === ALL_FLAG_VALUE) {
+    return allOffers;
+  }
+
+  if (flags[OFFER_FLAG_NAME] === undefined) {
+    return checkboxes<OfferFromProviderArtifacts, never>({
+      message: `Select one or more offer names from ${providerArtifactsConfig.$getPath()}`,
+      options: allOffers.map((offer) => {
+        return {
+          name: offer.offerName,
+          value: offer,
+        };
+      }),
+      validate: (choices: string[]) => {
+        if (choices.length === 0) {
+          return "Please select at least one offer name";
+        }
+
+        return true;
+      },
+      oneChoiceMessage(choice) {
+        return `One offer found at ${providerArtifactsConfig.$getPath()}: ${color.yellow(
+          choice,
+        )}. Do you want to select it`;
+      },
+      onNoChoices() {
+        commandObj.error(
+          `You must have at least one offer specified in ${providerArtifactsConfig.$getPath()}`,
+        );
+      },
+      flagName: OFFER_FLAG_NAME,
+    });
+  }
+
+  const [notFoundOffers, foundOffers] = splitErrorsAndResults(
+    commaSepStrToArr(flags[OFFER_FLAG_NAME]),
     (offerName) => {
-      const offer = providerArtifactsConfig.offers[offerName];
+      const offer = allOffers.find((o) => {
+        return o.offerName === offerName;
+      });
 
       if (offer === undefined) {
         return { error: offerName };
       }
 
-      return { result: { offerName, ...offer } };
+      return { result: offer };
     },
   );
 
   if (notFoundOffers.length > 0) {
     commandObj.error(
-      `Offers not found in ${providerArtifactsConfig.$getPath()} 'offer' property: ${notFoundOffers.join(
-        ", ",
-      )}`,
+      `Offer${notFoundOffers.length === 1 ? "" : "s"} ${color.yellow(
```

Comment:
Let's have additional util function for that, e.g. `pluralize`?
It will take word and number. And depending on passed number, the function will return singular or plural form of the word. 
examples: `pluralize("Offer", 1)// Offer`, `pluralize("Cosmos", 2)// Cosmoses`

Reply:
yeah, I thought about that, maybe something like that can be used, added. Mostly I try to avoid these situations, but here I couldn't come up with an alternative sentence that sounds good and doesn't have this issue